src/Db/PdoRepository.php

```diff
@@ -302,4 +309,175 @@ public function truncateWatches()
+    {
+        return [
+            'where' => $this->buildWhere($options['conditions']),
+            'sort' => $this->buildSort($options),
+            'direction' => $this->buildDirection($options),
+            'perPage' => $options['perPage'] ?? SearcherInterface::DEFAULT_PER_PAGE,
+        ];
+    }
+
+    /**
+     * build pdo where
+     *
+     * @param array $search
+     *
+     * @return array
+     */
+    public function buildWhere(array $search): array
+    {
+        $where = ['conditions' => '1=1', 'params' => []];
+
+        if (empty($search)) {
+            return $where;
+        }
+
+        if (!empty($search['limit_custom']) && $search['limit_custom'][0] === 'P') {
+            $search['limit'] = $search['limit_custom'];
+        }
+        $hasLimit = (!empty($search['limit']) && $search['limit'] != -1);
+
+        // simple_url equals match
+        if (isset($search['simple_url'])) {
+            $where['conditions'] .= ' and simple_url = :simple_url';
+            $where['params']['simple_url'] = $search['simple_url'];
+        }
+
+        if (!empty($search['date_start']) && !$hasLimit) {
+            $where['conditions'] .= ' and request_date >= :date_start';
+            $where['params']['date_start'] = $search['date_start'];
+        }
+
+        if (!empty($search['date_end']) && !$hasLimit) {
+            $where['conditions'] .= ' and request_date <= :date_end';
+            $where['params']['date_end'] = $search['date_end'];
+        }
+
+        if (!empty($search['request_start'])) {
+            $where['conditions'] .= ' and request_ts >= :request_start';
+            $where['params']['request_start'] = strtotime($search['request_start']);
+        }
+
+        if (!empty($search['request_end'])) {
+            $where['conditions'] .= ' and request_ts <= :request_end';
+            $where['params']['request_end'] = strtotime($search['request_end']);
+        }
+
+        // TODO need JSON support
+        if (!empty($search['remote_addr'])) {
+            $where['conditions'] .= ' and SERVER like :remote_addr';
+            $where['params']['remote_addr'] = '%' . $search['remote_addr'] . '%';
+        }
+
+        if (isset($search['cookie'])) {
+            $where['conditions'] .= ' and SERVER like :cookie';
+            $where['params']['cookie'] = '%' . $search['cookie'] . '%';
+        }
+
+        if (!empty($search['server_name'])) {
+            $where['conditions'] .= ' and SERVER like :server_name';
+            $where['params']['server_name'] = '%' . $search['server_name'] . '%';
+        }
+
+        if ($hasLimit && $search['limit'][0] === 'P') {
+            $date = new DateTime();
+            try {
+                $date->sub(new DateInterval($search['limit']));
+                $where['conditions'] .= ' and request_ts >= :limit_start';
+                $where['params']['limit_start'] = $date->getTimestamp();
+            } catch (\Exception $e) {
```

Comment:
catch more specific exception

Reply:
DateInterval only throws this kind of exception